test/trento/sap_systems/sap_system_test.exs

```diff
@@ -1478,6 +1593,135 @@ defmodule Trento.SapSystems.SapSystemTest do
       end
     end
 
+    test "should always deregister the SAP system when the deregistration command is received" do
+      sap_system_id = UUID.uuid4()
+      deregistered_at = DateTime.utc_now()
+
+      message_server_host_id = UUID.uuid4()
+      abap_host_id = UUID.uuid4()
+      enqrep_host_id = UUID.uuid4()
+
+      message_server_instance_number = "01"
+      abap_instance_number = "02"
+      enqrep_server_instance_number = "03"
+
+      application_sid = fake_sid()
+
+      assert_events_and_state(
+        [
+          build(
+            :application_instance_registered_event,
+            sap_system_id: sap_system_id,
+            features: "MESSAGESERVER|ENQUE",
+            sid: application_sid,
+            host_id: message_server_host_id,
+            instance_number: message_server_instance_number
+          ),
+          build(
+            :application_instance_registered_event,
+            sap_system_id: sap_system_id,
+            features: "ABAP|GATEWAY|ICMAN|IGS",
+            sid: application_sid,
+            host_id: abap_host_id,
+            instance_number: abap_instance_number
+          ),
+          build(
+            :sap_system_registered_event,
+            sap_system_id: sap_system_id,
+            sid: application_sid
+          ),
+          build(
```

Comment:
We can remove this last `build`, it just makes the test longer without adding anything

Reply:
done